src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -1087,8 +1087,16 @@ void RNN::prepareParams() {
             DnnlExtensionUtils::makeDescriptor(query_weights_md(1)),
             DnnlExtensionUtils::makeDescriptor(query_weights_md(2))
         };
-        prepareMemory(intDescs);
-        wasMemoryPrepared = true;
+
+        if (intDescs[1]->getFormatKind() == dnnl_format_kind_rnn_packed) {
+            // Need to update its weights memory format of rnn_packed as the primitive gets updated.
+            wFormatWasChanged = true;
+        }
+
```

Comment:
A more robust method would be to implement logic to compare the queried new memory descriptors with the existing memory's exact descriptor (result generated by previous call to prepareMmeory) to see whether it needs to be prepared again, on per input basis. since we also observed that brgconv may require different weight layout when batched is changed.

Reply:
I was thinking of that too. 
All nodes, ASAIK, has similar logic of rnn and conv - it assumes the weights are prepared in memory as it indicts to primitive creation and won't change the weights along with the primitive instance itself. 
But in fact, any time when the primitive instance is changed, its inputs (and weights) memory format may change too.

Reply:
Let me know if you agree to change the logic too. @EgorDuplensky

Reply:
Comparing the descriptors is seemed to be a correct way of doing this.
The only problem here is that it can will affect performance.

Reply:
Yes, that's my concern too.  Since so far we only see it in rnn-packed memory format, that's why I only check format-kind instead of comparing the descriptors for all cases.

Reply:
Sorry, I was wrong, only `dnnl_primitive_desc_query_md()` is done unconditionally, the heavy work `prepareMemory()` is only done on `dnnl_format_kind_rnn_packed` weight format which only appears on non-brgemm implementations, since brgemm becomes overwhelming implementation right now, maybe we can accept a little overhead here, but I'm 100% sure since there is still WA to prevent using brgemm in some cases.